lib/postcode_validation/gateway/pca_address_list.rb

```diff
@@ -12,8 +12,16 @@ def query(search_term:, country:, more_results_id: nil)
         address_list_for_postcode(country, search_term, more_results_id).map do |row|
           raise PCARequestError, error_message(row) if row.key?('Error')
 
-          PostcodeValidation::Domain::Address.new(row: row)
-        end
+          if row['Type'] == 'Postcode'
+            self.class.new.query(
```

Comment:
Do we want to control the amount of recursing? Is this list gonna be a lot?

Wonder if this is a business requirement or an adapter requirement

i.e. the amount of recursing, who would want to adjust the amount of results pulled back? Is it the 
Actor: PCA Predict, or,
Actor: I'm searching for addresses

This would influence whether the use case class needs to know about the setting this or whether the gateway would take ownership of controlling this